scripts/paper/aggregate_stwo_phase30_source_bound_manifest_reuse_benchmark.py

```
@@ -0,0 +1,182 @@
+    benchmark_version = None
+    semantic_scope = None
+    timing_unit = None
+    canonical_rows: list[dict[str, Any]] | None = None
+    timing_samples: dict[tuple[str, str, int], list[float]] = {}
+
+    for input_path in args.inputs:
+        payload = json.loads(input_path.read_text(encoding="utf-8"))
+        if payload.get("timing_mode") != "measured_single_run":
+            raise SystemExit(
+                f"{input_path} must be a measured single-run benchmark payload; got {payload.get('timing_mode')!r}"
+            )
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)